mewarpx/mewarpx/utils_store/runsummary.py

```diff
@@ -0,0 +1,38 @@
+"""
+Utility class for printint out performance information after a run
+"""
+
+import time
+
+class RunSummary(object):
+    def __init__(self):
+        from pywarpx import callbacks
+        self.start_time = time.time()
+        self.prev_time = self.start_time
+        self.particle_steps_total = 0
+        self.total_particle_step_rate = 0
+        self.previous_particle_steps_total = 0
+        callbacks.installafterstep(self.update_particle_steps)
+
+    def update_particle_steps(self):
+        from mewarpx.mwxrun import mwxrun
+        num_particles = mwxrun.get_npart()
+        self.particle_steps_total += num_particles
+
+    def print_performance_summary(self):
+        from pywarpx import _libwarpx
+        total_time = time.time() - self.start_time
+        total_timesteps = _libwarpx.libwarpx.warpx_getistep(0)
+        steps_per_second = total_timesteps / total_time
+
+        n_procs = _libwarpx.libwarpx.warpx_getNProcs()
+        steps_per_second_per_proc = steps_per_second / n_procs
+
+        particle_steps_per_second = self.particle_steps_total / total_time
+        particle_steps_per_second_per_proc = particle_steps_per_second / n_procs
+        print("\n### Run Summary ###")
+
+        print(f"steps / second : {steps_per_second:.4f}")
+        print(f"steps / second / proc : {steps_per_second_per_proc:.4f}")
+        print(f"particle * steps / second : {particle_steps_per_second:.4f}")
+        print(f"particle * steps / second / proc : {particle_steps_per_second_per_proc:.4f}")
```

Comment:
Should these use the logger as well or is there a reason to print them instead?

Reply:
Haha yeah they probably should use the logger